status-whisper-usage-spec.md

```diff
@@ -59,9 +60,41 @@ Whisper's Proof Of Work algorithm is used to deter denial of service and various
 * proof-of-work requirement not larger than `0.002`
 * time-to-live not lower than `10` (in seconds)
 
-<!-- TODO: provide an instruction how to start a Whisper node with proper configuration using geth.-->
+## Handshake
 
-<!-- @TODO: is there a higher bound -->
+Handshake is a RLP-encoded packet sent to a newly connected peer. It MUST start with a Status Code (`0x00`) and follow up with items:
+```
+[ protocolVersion, PoW, bloom, isLightNode, confirmationsEnabled, rateLimits ]
+```
+
+`protocolVersion`: version of the Whisper protocol
+`PoW`: minimum PoW accepted by the peer
+`bloom`: bloom filter of Whisper topic accepted by the peer
+`isLightNode`: when true, the peer won't forward messages
+`confirmationsEnabled`: when true, the peer will send message confirmations
+`rateLimits`: is `[ RateLimitIP, RateLimitPeerID, RateLimitTopic ]` where each values is an integer with a number of accepted packets per second per IP, Peer ID, and Topic respectively
+
```

Comment:
```suggestion

`bloom, isLightNode, confirmationsEnabled, and rateLimits` are all optional arguments in the handshake. However, if you specify optional field you MUST also specify all optional fields preceding it, in order to be unambiguous.
```

Reply:
This was my initial suggestion and it is stricter but also maintains compatibility with EIP 627. @kdeme @adambabik do you see any issues with this? Or would you definitely prefer an even stricter version, where all fields are always required (thus breaking compatibility with EIP-627)?

Reply:
(Right now it is ambiguous because current EIP627 makes bloom filter optional, and this spec diff also mentions "optional rate limiting". Also typo in including 'and' in above suggestion quoted form)

Reply:
I don't see any issues. I don't think there are any clients that specify some optional field without others.